File: backend/app/schemas/diagnose.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class ModernTreatment(BaseModel):
    chemical: str | None
    dosage: str | None
    cost_per_acre: float | None
    supply_note: str | None


class IndigenousRemedy(BaseModel):
    name: str | None
    method: str | None
    preparation_ta: str | None


class TreatmentOut(BaseModel):
    modern: ModernTreatment
    indigenous: IndigenousRemedy


class DiseaseOut(BaseModel):
    id: str
    name_en: str
    name_ta: str


class DiagnoseResponse(BaseModel):
    diagnosis_id: uuid.UUID
    disease: DiseaseOut | None = None
    confidence: float | None = None
    confidence_level: str | None = None  # "high" | "medium" | "low"
    heatmap_url: str | None = None
    shap_label_ta: str | None = None
    treatment: TreatmentOut | None = None
    low_confidence_prompt_ta: str | None = None
    low_confidence_prompt_en: str | None = None


class DiagnoseHistoryItem(BaseModel):
    diagnosis_id: uuid.UUID
    disease_name_ta: str | None
    disease_name_en: str | None
    confidence: float | None
    created_at: datetime
    heatmap_url: str | None

    class Config:
        from_attributes = True


class DiagnoseHistoryResponse(BaseModel):
    diagnoses: list[DiagnoseHistoryItem]

File: frontend/src/types/diagnose.ts
export interface ModernTreatment {
  chemical: string | null
  dosage: string | null
  cost_per_acre: number | null
  supply_note: string | null
}

export interface IndigenousRemedy {
  name: string | null
  method: string | null
  preparation_ta: string | null
}

export interface TreatmentOut {
  modern: ModernTreatment
  indigenous: IndigenousRemedy
}

export interface DiseaseOut {
  id: string
  name_en: string
  name_ta: string
}

export interface SimilarDisease {
  name_en: string
  name_ta: string
  confidence: number
}

export interface SymptomOption {
  key: string
  label_en: string
  label_ta: string
}

export interface DiagnoseResponse {
  diagnosis_id: string
  disease: DiseaseOut | null
  confidence: number | null
  confidence_level: 'high' | 'medium' | 'low' | null
  source: 'ml_model' | 'symptom_match' | 'symptom_check_needed' | null
  heatmap_url: string | null
  shap_label_ta: string | null
  treatment: TreatmentOut | null
  low_confidence_prompt_ta: string | null
  low_confidence_prompt_en: string | null
  rejection_reason: 'image_quality' | 'similar_diseases' | null
  similar_diseases: SimilarDisease[] | null
  requires_symptom_check: boolean | null
  prompt_en: string | null
  prompt_ta: string | null
  symptoms_to_show: SymptomOption[] | null
  matched_symptoms: string[] | null
  match_score: number | null
}

export interface DiagnoseHistoryItem {
  diagnosis_id: string
  disease_name_ta: string | null
  disease_name_en: string | null
  confidence: number | null
  created_at: string
  heatmap_url: string | null
}

export interface DiagnoseHistoryResponse {
  diagnoses: DiagnoseHistoryItem[]
}
